Packs/FeedMandiant/Integrations/FeedMandiant/FeedMandiant.py

```diff
@@ -0,0 +1,692 @@
+    """
+    raw_indicator = {k: v for k, v in raw_indicator.items() if v and v != 'redacted'}  # filter none and redacted values
+    fields = {'primarymotivation': raw_indicator.get('motivations'),
+              'tags': [industry.get('name') for industry in raw_indicator.get('industries', [])],
+              'aliases': [alias.get('name') for alias in raw_indicator.get('aliases', [])],
+              'firstseenbysource': [item.get('earliest') for item in raw_indicator.get('observed', [])],
+              'lastseenbysource': [item.get('recent') for item in raw_indicator.get('observed', [])],
+              'targets': [target.get('name') for target in raw_indicator.get('locations', {}).get('target', [])],
+              'stixid': raw_indicator.get('id'),
+              'name': raw_indicator.get('name'),
+              'description': raw_indicator.get('description'),
+              'updateddate': raw_indicator.get('last_updated'),
+              }
+
+    fields = {k: v for k, v in fields.items() if v and v != 'redacted'}  # filter none and redacted values
+    indicator_obj = {
+        'value': raw_indicator.get('name'),
+        'type': ThreatIntel.ObjectsNames.THREAT_ACTOR,
+        'rawJSON': raw_indicator,
+        'score': get_verdict(raw_indicator.get('mscore')),
+        'fields': fields,
+        'relationships': get_relationships_actor(raw_indicator)
+    }
+    return indicator_obj
+
+
+def create_indicator(raw_indicator: Dict) -> Dict:
+    """
+    Create indicator
+    Args:
+        raw_indicator (Dict): raw indicator
+
+    Returns: Parsed indicator
+    """
+    fields = {'primarymotivation': raw_indicator.get('motivations'),
+              'firstseenbysource': raw_indicator.get('first_seen'),
+              'lastseenbysource': raw_indicator.get('last_seen'),
+              'stixid': raw_indicator.get('id'),
+              }
+
+    fields = {k: v for k, v in fields.items() if v and v != 'redacted'}  # filter none and redacted values
+    indicator_obj = {
+        'value': raw_indicator.get('value'),
+        'type': MAP_INDICATORS_TYPE[raw_indicator.get('type', '')],
+        'rawJSON': raw_indicator,
+        'score': get_verdict(raw_indicator.get('mscore')),
+        'fields': fields
+    }
+    return indicator_obj
+
+
+MAP_INDICATORS_FUNCTIONS = {
+    'Malware': create_malware_indicator,
+    'Actors': create_actor_indicator,
+    'Indicators': create_indicator
+}
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: MandiantClient) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client._generate_token()
```

Comment:
Instead of just generating the token let's call the fetch_indicators flow with a flag that will sugnal we don't want to update the context (see my general comments). This way the user can really check that all params are valid and working.